onnxruntime/test/providers/kernel_def_hash_test.cc

```diff
@@ -127,7 +127,11 @@ void CheckKernelDefHashes(const KernelDefHashes& actual, const KernelDefHashes&
         "Some expected kernel def hashes were not found.\n",
         kNoteReference, "\n",
         DumpKernelDefHashes(expected_minus_actual));
-    ADD_FAILURE() << message;
+    if (is_strict) {
```

Comment:
now if is_strict is false, this function only verifies that the inputs are sorted, which is not much.
the "strict" behavior is described here and would need to be updated if it changes
https://github.com/microsoft/onnxruntime/blob/5ad6dbb314f3a2dd10450273ceeeb0513cad79b3/onnxruntime/test/providers/kernel_def_hash_test.cc#L90-L92